### PR TITLE
feat: Add agent skills feature based on Claude skill guidelines

### PR DESCRIPTION
**Pull request by lannerate**
Implement a comprehensive skills system for extending agent capabilities with specialized knowledge and behaviors.

## Core Components
- Added Skill data model with validation (app/schema_skill.py)
- Implemented SkillManager for discovery and loading (app/skill_manager.py)
- Extended config with SkillsSettings (app/config.py)
- Integrated skills into Manus agent execution (app/agent/manus.py)

## Features
- Automatic skill activation based on relevance scoring
- Multi-file skill support with progressive disclosure
- Tool restrictions per skill
- Configurable search paths and thresholds
- Manual skill activation/deactivation
- CLI integration (--list-skills command)

## Example Skills
- code-review: Code quality and security reviews
- documentation: Technical documentation writing
- debugging: Systematic debugging approach

## Documentation
- SKILLS.md: Complete user guide
- SKILLS_IMPLEMENTATION.md: Technical details
- SKILLS_TESTING.md: Test verification report
- ARCHITECTURE.md: Updated system overview

## Testing
- Comprehensive test suite (test_skills.py)
- All 7 tests passing
- Integration tests verified
- Edge cases handled

Refs: https://code.claude.com/docs/en/skills

**Features**

- Feature 1
- Feature 2

**Feature Docs**

**Influence**

**Result**

**Other**

### CODE REVIEW

File: test_skills.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+    for name, skill in skills.items():
+        print(f"  - {name}: {skill.description[:60]}...")
+
+    assert len(skills) == 3, f"Expected 3 skills, got {len(skills)}"
+    assert "code-review" in skills, "code-review skill not loaded"
+    assert "documentation" in skills, "documentation skill not loaded"
+    assert "debugging" in skills, "debugging skill not loaded"
+
+    print("\n✅ Test 1 PASSED: All skills loaded correctly")
```
Comment (Mrart) [inline test_skills.py:36]: It seem that we not need this green tag？
